src/ORM/ResultSet.php

```diff
@@ -29,6 +29,9 @@
  * This object is responsible for correctly nesting result keys reported from
  * the query, casting each field to the correct type and executing the extra
  * queries required for eager loading external associations.
+ *
+ * @template T
```

Comment:
Changing this to `@template T of \Cake\Datasource\EntityInterface|array` and replacing `object` with `\Cake\Datasource\EntityInterface` in all union types should make the static analyzers happy. The `object|array` has already been changed to `\Cake\Datasource\EntityInterface|array` in 5.x.

Reply:
I am definitely not that accostumed to this kind of phpdoc but I think I got it now 😅